### PR TITLE
!feat(api/gateway): return txdata from submitPFB if it is present

### PR DESCRIPTION
**Pull request by walldiss**
## Overview

Closes https://github.com/celestiaorg/celestia-node/issues/2328

### CODE REVIEW

File: api/gateway/state.go [hunk 1]
```diff
@@ -149,16 +156,27 @@ func (h *Handler) handleSubmitPFB(w http.ResponseWriter, r *http.Request) {
...
+	if err != nil && txResp == nil {
+		// no tx data to return
 		writeError(w, http.StatusInternalServerError, submitPFBEndpoint, err)
-		return
 	}
-	resp, err := json.Marshal(txResp)
+
+	resp := submitPFBResponse{
+		Tx:    txResp,
+		Error: err.Error(),
```
Comment (Wondertan) [inline api/gateway/state.go:166]: We actually don't need to return an error so we can just return txResp. In SubmitPayForBlob we derive the error from the txResponse, so by returning the error we just send duplicate data.

Comment (walldiss) [reply api/gateway/state.go:166]: Ah, ok! TIL

Comment (Wondertan) [reply api/gateway/state.go:166]: > Kinda weird API
💯

Comment (walldiss) [reply api/gateway/state.go:166]: Remved redundant error

**Review by Wondertan — APPROVED** [review]

**Pull request merged**